assets/src/edit-story/components/library/mediaLibrary.js

```diff
@@ -30,41 +46,143 @@ const Title = styled.h3`
 	font-size: 19px;
 	line-height: 20px;
 	line-height: 1.4em;
-	flex: 2 0 0;
+	flex: 3 0 0;
 `;
 
 const Header = styled.div`
 	display: flex;
-	margin: 0px 4px 10px;
+	margin: 0px 0px 25px;
 `;
 
 const Message = styled.div`
-	 color: ${ ( { theme } ) => theme.colors.fg.v1 };
+	color: ${ ( { theme } ) => theme.colors.fg.v1 };
 	font-size: 19px;
 `;
 
+const FilterButtons = styled.div`
+	border-bottom:  2px solid ${ ( { theme } ) => theme.colors.mg.v1 };
+	padding: 18px 0px;
+	margin: 10px 0px 15px;
+`;
+
+const FilterButton = styled.button`
+	border: 0px;
+	background: none;
+	padding: 0px;
+	margin: 0px 28px 0px 0px;
+	color: ${ ( { theme } ) => theme.colors.mg.v1 };
+	font-size: 13px;
+	${ ( { active } ) => active && ` color: #fff; font-weight: bold; ` };
+`;
+
+const Search = styled.input`
+	width: 100%;
+	background: ${ ( { theme } ) => theme.colors.mg.v1 } !important;
+	border-color: ${ ( { theme } ) => theme.colors.mg.v1 } !important;
+	color: ${ ( { theme } ) => theme.colors.mg.v2 } !important;
+	padding: 2px 10px 2px 33px !important;
+	&::placeholder {
+	  color: ${ ( { theme } ) => theme.colors.mg.v2 };
+	}
+`;
+
+const SUPPORTED_IMAGE_TYPES = [
+	'image/png',
+	'image/jpeg',
+	'image/jpg',
+	'image/gif',
+];
+
+const SUPPORTED_VIDEO_TYPES = [
+	'video/mp4',
+];
+
+const FILTERS = [
+	{ filter: '', name: 'All' },
+	{ filter: 'image', name: 'Images' },
+	{ filter: 'video', name: 'Video' },
+];
+
+const DEFAULT_WIDTH = 150;
+
 function MediaLibrary( { onInsert } ) {
 	const {
-		state: { media, isMediaLoading, isMediaLoaded, mediaType },
-		actions: { loadMedia, setIsMediaLoading, setIsMediaLoaded },
+		state: { media, isMediaLoading, isMediaLoaded, mediaType, searchTerm },
```

Comment:
Will the same search state be shared between Shapes, Media, etc.? E.g. if I'd switch from Media Library to Shapes, would the search term typed in Media appear there? I'd expect the search term to remain for the Media Library and be empty in a new tab. The `searchTerm` seems state seems to be shared between the elements right now, might make sense to have it per-library basis so that it would persist in the Media Library but the same term not appear in the Shapes when switching. I suppose it can be addressed later, too, though, when implementing the search for other elements. Thoughts?

Reply:
I think we do need to think about this later, but good to flag it now. We can make a lot of this generate, like the style search field so it can be reused.